inventory_management_system_api/routers/v1/catalogue_category.py

```diff
@@ -117,6 +117,14 @@ def create_catalogue_category(
         catalogue_category = catalogue_category_service.create(catalogue_category)
         return CatalogueCategorySchema(**catalogue_category.model_dump())
     except (MissingRecordError, InvalidObjectIdError) as exc:
+        if (
+            catalogue_category.catalogue_item_properties is not None
+            and any(str(property.unit_id) in str(exc) for property in catalogue_category.catalogue_item_properties)
+        ) or "units" in str(exc).lower():
+            message = "The specified unit does not exist"
+            logger.exception(message)
+            raise HTTPException(status_code=status.HTTP_404_NOT_FOUND, detail=message) from exc
```

Comment:
```suggestion
            raise HTTPException(status_code=status.HTTP_422_UNPROCESSABLE_ENTITY, detail=message) from exc
```
We use this elsewhere if it can't process it due to the missing value, I believe 404's are more for the route parameters themselves rather than the body.